src/ExportMap.js

```diff
@@ -305,7 +306,7 @@ ExportMap.get = function (source, context) {
 ExportMap.for = function (context) {
   const { path } = context;
 
-  const cacheKey = hashObject(context).digest('hex');
+  const cacheKey = context.cacheKey || hashObject(context).digest('hex');
```

Comment:
Trying not to change the public interface of `ExportMap.for`. In case we did not pre-calculate a cacheKey, it will behave like before.